SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m

```diff
@@ -17,6 +17,510 @@ static void FreeImageData(void *info, const void *data, size_t size) {
+            sRGB = CreateColorSpaceMono(colorPrimaries, transferCharacteristics);
+            if(sRGB == NULL) {
+                sRGB = defaultColorSpace;
+            }
+        });
+        *ref = sRGB;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(colorPrimaries == AVIF_NCLX_COLOUR_PRIMARIES_BT709 &&
+       transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_BT709) {
+        static CGColorSpaceRef bt709 = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            bt709 = CreateColorSpaceMono(colorPrimaries, transferCharacteristics);
+            if(bt709 == NULL) {
+                bt709 = defaultColorSpace;
+            }
+        });
+        *ref = bt709;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(colorPrimaries == AVIF_NCLX_COLOUR_PRIMARIES_BT2020 &&
+       (transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_BT2020_10BIT ||
+        transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_BT2020_12BIT)) {
+        static CGColorSpaceRef bt2020 = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            bt2020 = CreateColorSpaceMono(colorPrimaries, transferCharacteristics);
+            if(bt2020 == NULL) {
+                bt2020 = defaultColorSpace;
+            }
+        });
+        *ref = bt2020;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(colorPrimaries == AVIF_NCLX_COLOUR_PRIMARIES_P3 &&
+       transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_SRGB) {
+        static CGColorSpaceRef p3 = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            p3 = CreateColorSpaceMono(colorPrimaries, transferCharacteristics);
+            if(p3 == NULL) {
+                p3 = defaultColorSpace;
+            }
+        });
+        *ref = p3;
+        *shouldRelease = FALSE;
+        return;
+    }
+
+    *ref = CreateColorSpaceMono(colorPrimaries, transferCharacteristics);
+    if(*ref != NULL) {
+        *shouldRelease = TRUE;
+    } else {
+        *ref = defaultColorSpace;
+        *shouldRelease = FALSE;
+    }
+}
+
+static void CalcColorSpaceRGB(avifImage * avif, CGColorSpaceRef* ref, BOOL* shouldRelease) {
+    static CGColorSpaceRef defaultColorSpace = NULL;
+    {
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            defaultColorSpace = CGColorSpaceCreateDeviceRGB();
+        });
+    }
+    if(avif->profileFormat == AVIF_PROFILE_FORMAT_NONE) {
+        *ref = defaultColorSpace;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(avif->profileFormat == AVIF_PROFILE_FORMAT_ICC) {
+        if(avif->icc.data && avif->icc.size) {
+            if(@available(macOS 10.12, iOS 10.0, tvOS 10.0, *)) {
+                *ref = CGColorSpaceCreateWithICCData(avif->icc.data);
+                *shouldRelease = TRUE;
+                return;
```

Comment:
call CGColorSpaceCreateWithICCProfile for iOS 8~9

Reply:
OK. I called it.